### PR TITLE
Fix Pretty JSON

### LINKED ISSUE #132: Pretty printing produces invalid json
Example:

```java
import io.avaje.jsonb.Json;
import io.avaje.jsonb.Jsonb;

import java.util.List;

@Json
public record PrettyExample(String name, List<String> values, int count) {

    public static void main(String[] args) {
        var jsonb = Jsonb.builder().build();
        var type = jsonb.type(PrettyExample.class);
        var example = new PrettyExample("test", List.of("one", "two"), 2);
        System.out.println("Normal output is correct: " + type.toJson(example));
        System.out.println("Pretty print output is invalid json and oddly formatted: " + type.toJsonPretty(example));
        System.out.println("Somehow invalid json can be parsed: " + type.fromJson(type.toJsonPretty(example)));
    }

}
```

Produces:

```
Normal output is correct: {"name":"test","values":["one","two"],"count":2}
Pretty print output is invalid json and oddly formatted: {
  "name": "test",
  "values":   [
"one","two"    ]

  "count": 2
}
Somehow invalid json can be parsed: PrettyExample[name=test, values=[one, two], count=2]
```

Note that in addition to producing invalid json, the formatting is odd, and it can somehow be round tripped.

### PR DESCRIPTION
**Pull request by SentryMan**
Fixes #132

### CODE REVIEW

File: jsonb/src/main/java/io/avaje/jsonb/stream/JGenerator.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ final class JGenerator implements JsonGenerator {
   private static final byte[] NULL = "null".getBytes(StandardCharsets.UTF_8);
   private static final byte[] TRUE = "true".getBytes(StandardCharsets.UTF_8);
   private static final byte[] FALSE = "false".getBytes(StandardCharsets.UTF_8);
-  private static final byte[] INDENT = "  ".getBytes(StandardCharsets.UTF_8);
+  private static final byte[] INDENT = "    ".getBytes(StandardCharsets.UTF_8);
```
Comment (rbygrave) [inline jsonb/src/main/java/io/avaje/jsonb/stream/JGenerator.java:40]: Why push this out to 4?

Comment (SentryMan) [reply jsonb/src/main/java/io/avaje/jsonb/stream/JGenerator.java:40]: that's how my ide formats json so I just went with it

Comment (SentryMan) [reply jsonb/src/main/java/io/avaje/jsonb/stream/JGenerator.java:40]: you can change if you like

Comment (rbygrave) [reply jsonb/src/main/java/io/avaje/jsonb/stream/JGenerator.java:40]: Yeah, it's in the area of "breaking change" for any asserting on json strings (which is a little risky I know). I'll put it back for now and keep pondering ...

**Pull request merged**